build: delete cmake code

dundargoc opened the pull request:
- build: remove unused variable CMAKE_C_COMPILER_ARG1

It was set in file cmake/i386-linux-gnu.toolchain.cmake which has been
removed since we don't use Travis anymore.

- build: remove code for cross-compilation

We don't support cross-compilation at the moment, so these can be safely
removed.

- build: remove url for 32-bit winyank

We don't support 32bit Windows anymore, so it's not needed. Also remove
TargetArch.cmake and related code as we don't need architecture
detection for the same reason.

- build: remove unnecessary translation-related code

The commands run in cmake script mode (-P) can simply be run in the main
cmake run instead.

- build: rely on builtin cmake downloading rather than custom script

DownloadAndExtractFile.cmake was initially introduced as a workaround to
avoid the massive amounts of logs generated by the download progress.
This is not a problem anymore as ExternalProject_Add has had the
DOWNLOAD_NO_PROGRESS option since cmake version 3.1.

- build: define EP_PREFIX property

This is just to avoid the boilerplate of definining PREFIX for
each dependency.

dundargoc committed:
build: remove unused variable CMAKE_C_COMPILER_ARG1

It was set in file cmake/i386-linux-gnu.toolchain.cmake which has been
removed since we don't use Travis anymore.

dundargoc committed:
build: remove code for cross-compilation

We don't support cross-compilation at the moment, so these can be safely
removed.

dundargoc committed:
build: remove url for 32-bit winyank

We don't support 32bit Windows anymore, so it's not needed. Also remove
TargetArch.cmake and related code as we don't need architecture
detection for the same reason.

justinmk commented on cmake.deps/cmake/UnibiliumCMakeLists.txt:
```diff
@@ -6,16 +6,7 @@ add_library(unibilium ${SRC_FILES})
-else()
-  set(TERMINFO_DIRS "\"\"")
-endif()
+set(TERMINFO_DIRS "\"\"")
```
> Remove non-windows code from dependencies if they're only used in Windows.

Aren't these "bundled" deps which could be used on any system?

dundargoc replied:
They could be, but they aren't. On `BuildUnibilium.cmake` we see that `UnibiliumCMakeLists.txt` is only used on Windows. But still, maybe it's safer to leave these files untouched in case we start use them for non-windows systems in the future?

justinmk replied:
That seems like a confusing bit of knowledge to implicitly encode in our bundled builds. (Side note: does it make sense for us to add CMakeLists.txt to https://github.com/neovim/unibilium ? )

justinmk commented on cmake.deps/cmake/BuildLibtermkey.cmake:
```diff
@@ -4,8 +4,6 @@ if(WIN32)
     COMMAND ${CMAKE_COMMAND} ${DEPS_BUILD_DIR}/src/libtermkey
       -DCMAKE_INSTALL_PREFIX=${DEPS_INSTALL_DIR}
-      # Pass toolchain
-      -DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN}
```
Just to confirm, does the blame / git log show that these TOOLCHAIN lines were added in a commit related to cross-compiling?

dundargoc replied:
The git blame doesn't really show anything other than a generic "added dependency". But toolchains are for cross-compilation. Also we don't have any toolchains to pass anymore since they were deleted.

dundargoc committed:
build: remove unnecessary translation-related code

The commands run in cmake script mode (-P) can simply be run in the main
cmake run instead.

dundargoc committed:
build: rely on builtin cmake downloading rather than custom script

DownloadAndExtractFile.cmake was initially introduced as a workaround to
avoid the massive amounts of logs generated by the download progress.
This is not a problem anymore as ExternalProject_Add has had the
DOWNLOAD_NO_PROGRESS option since cmake version 3.1.

dundargoc commented:
@carlocab `DownloadAndExtractFile.cmake` has a Homebrew-specific fix from a while back that this PR will remove. I was wondering if you wouldn't mind taking a quick look whenever you have time and confirm whether it's still required or not? Here's the relevant code section:

```cmake
# Check whether the source has been downloaded. If true, skip it.
# Useful for external downloads like homebrew.
if(USE_EXISTING_SRC_DIR)
  if(EXISTS "${SRC_DIR}" AND IS_DIRECTORY "${SRC_DIR}")
    file(GLOB EXISTED_FILES "${SRC_DIR}/*")
    if(EXISTED_FILES)
      message(STATUS "${SRC_DIR} is found and not empty, skipping download and extraction. ")
      return()
    endif()
  endif()
  message(FATAL_ERROR "USE_EXISTING_SRC_DIR set to ON, but '${SRC_DIR}' does not exist or is empty.")
endif()
```

I'm pretty sure the default cmake behavior is to skip downloading if the files exist but I don't want to break anything :).

carlocab commented:
Thanks for the ping! Let me have a look later today. I don't think we rely on this anymore, but let me test it out to make sure.

dundargoc commented:
Aside from the potential Homebrew-specific code that was added (which I'll defer to @carlocab), other regressions in behavior for removing `DownloadAndExtractFile.cmake`:

- Warn user if their cmake wasn't built with SSL support.
- Allow skipping hash check when downloading dependencies.
- Extra download retries for libtermkey as download failures were apparently common. Not sure how common this is nowadays.

Only the hash check skip seems like it could be worth preserving, but I'm fine not having that either personally.

dundargoc committed:
build: define EP_PREFIX property

This is just to avoid the boilerplate of definining PREFIX for
each dependency.

dundargoc commented on cmake.deps/CMakeLists.txt:
```diff
@@ -142,6 +142,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
 endif()
 
 include(ExternalProject)
+set_directory_properties(PROPERTIES EP_PREFIX "${DEPS_BUILD_DIR}")
```
I can revert this if people think defining `PREFIX` for each dependency is more explicit/clearer.

carlocab commented:
> @carlocab `DownloadAndExtractFile.cmake` has a Homebrew-specific fix from a while back that this PR will remove. I was wondering if you wouldn't mind taking a quick look whenever you have time and confirm whether it's still required or not? Here's the relevant code section:
> 
> ```cmake
> # Check whether the source has been downloaded. If true, skip it.
> # Useful for external downloads like homebrew.
> if(USE_EXISTING_SRC_DIR)
>   if(EXISTS "${SRC_DIR}" AND IS_DIRECTORY "${SRC_DIR}")
>     file(GLOB EXISTED_FILES "${SRC_DIR}/*")
>     if(EXISTED_FILES)
>       message(STATUS "${SRC_DIR} is found and not empty, skipping download and extraction. ")
>       return()
>     endif()
>   endif()
>   message(FATAL_ERROR "USE_EXISTING_SRC_DIR set to ON, but '${SRC_DIR}' does not exist or is empty.")
> endif()
> ```
> 
> I'm pretty sure the default cmake behavior is to skip downloading if the files exist but I don't want to break anything :)

Should be good to 🔥

dundargoc commented:
Great. Thanks for the help!

Yup, can't forget about testing for BSD :)
